contracts/stellar-interchain-token-service/src/types.rs

```diff
@@ -27,8 +27,17 @@ pub struct DeployInterchainToken {
 
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct RegisterTokenMetadata {
-    pub token_address: Bytes,
     pub decimals: u8,
+    pub token_address: Bytes,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct LinkToken {
+    pub token_id: BytesN<32>,
+    pub token_manager_type: i128,
```

Comment:
there is a Uint256 type
https://docs.rs/soroban-sdk/latest/soroban_sdk/xdr/struct.Uint256.html

However, used an enum instead?
```suggestion
    pub token_manager_type: TokenManagerType,
```